core/services/registrysyncer/syncer.go

```diff
@@ -291,7 +416,10 @@ func (s *registrySyncer) AddLauncher(launchers ...Launcher) {
 
 func (s *registrySyncer) Close() error {
 	return s.StopOnce("RegistrySyncer", func() error {
+		s.mu.Lock()
+		defer s.mu.Unlock()
 		close(s.stopCh)
```

Comment:
I think this should be before the lock, because it can cause whoever is holding the lock to yield faster, and short circuit anyone who gets the lock first.